### PR TITLE
Reduce connections used by MockNioTransport

### PR DESCRIPTION
**Pull request by Tim-Brooks**
The MockNioTransport (similar to the MockTcpTransport) is used for integ
tests. The MockTcpTransport has always only opened a single for all of
its work. The MockNioTransport has awlays opened the default number of
connections (13). This means that every test where two transports
connect requires 26 connections. This is more than is necessary. This
commit modifies the MockNioTransport to only require 3 connections.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/transport/nio/MockNioTransport.java [hunk 1]
```diff
@@ -128,6 +132,34 @@ protected void stopInternal() {
         profileToChannelFactory.clear();
     }
 
+    @Override
+    protected ConnectionProfile resolveConnectionProfile(ConnectionProfile connectionProfile) {
+        ConnectionProfile connectionProfile1 = resolveConnectionProfile(connectionProfile, defaultConnectionProfile);
```
Comment (jasontedor) [inline test/framework/src/main/java/org/elasticsearch/transport/nio/MockNioTransport.java]: Can you give it a more meaningful name?

File: test/framework/src/main/java/org/elasticsearch/transport/nio/MockNioTransport.java [hunk 2]
```diff
@@ -128,6 +132,34 @@ protected void stopInternal() {
...
+        if (connectionProfile1.getNumConnections() <= 3) {
+            return connectionProfile1;
+        }
+        ConnectionProfile.Builder builder = new ConnectionProfile.Builder();
+        Set<TransportRequestOptions.Type> allTypesWithConnection = new HashSet<>();
+        Set<TransportRequestOptions.Type> allTypesWithoutConnection = new HashSet<>();
+        for (TransportRequestOptions.Type type :TransportRequestOptions.Type.values()) {
```
Comment (jasontedor) [inline test/framework/src/main/java/org/elasticsearch/transport/nio/MockNioTransport.java]: Nit: space after the `:`

File: test/framework/src/test/java/org/elasticsearch/transport/nio/SimpleMockNioTransportTests.java [hunk 3]
```diff
@@ -94,6 +94,10 @@ protected MockTransportService build(Settings settings, Version version, Cluster
         return transportService;
     }
 
+    protected int channelsPerNodeConnection() {
```
Comment (jasontedor) [inline test/framework/src/test/java/org/elasticsearch/transport/nio/SimpleMockNioTransportTests.java]: This appears unused?

**Review by jasontedor — CHANGES_REQUESTED** [review]
I left a few comments.

**Comment by tbrooks8** [comment]
@jasontedor - I made changes based on your review. `channelsPerNodeConnection` is used. But I added the `@Override` to make that clear.

**Comment by jasontedor** [comment]
> `channelsPerNodeConnection` is used. But I added the `@Override` to make that clear.

Great, thanks.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Pull request merged**